static_docs/class.md

```diff
@@ -0,0 +1,143 @@
+# Class
+
+In Ruby, a `class` is a blueprint for creating objects that share similar attributes and behaviors. Classes encapsulate data and methods, following object-oriented programming principles.
+
+```ruby
+# Basic class definition
+class Person
+  def initialize(name)
+    @name = name
+  end
+
+  def greet
+    puts "Hello, #{@name}!"
+  end
+end
+
+person = Person.new("Ruby")
+person.greet
+# Output:
+# Hello, Ruby!
+```
+
+Classes can include instance methods, class methods, and various types of variables.
+
+```ruby
+class Product
+  # rubocop:disable Style/ClassVars
+  # Class variable (shared across all instances)
+  @@count = 0
+  # rubocop:enable Style/ClassVars
+
+  # rubocop:disable Style/ClassMethodsDefinitions
+  # Class method
+  def self.count
+    @@count
+  end
+  # rubocop:enable Style/ClassMethodsDefinitions
+
+  def initialize(name, price)
+    @name = name
+    @price = price
+    # rubocop:disable Style/ClassVars
+    @@count += 1
+    # rubocop:enable Style/ClassVars
+  end
+
+  # Instance method
+  def details
+    "#{@name}: $#{@price}"
+  end
+end
+
+book = Product.new("Ruby Guide", 29.99)
+puts Product.count # Output: 1
+puts book.details # Output: Ruby Guide: $29.99
+```
+
+## Inheritance
+
+Classes can inherit behavior from other classes using the `<` operator. A class can only inherit from one parent class.
+
+```ruby
+# Parent class
+class Animal
+  def speak
+    "Some sound"
+  end
+end
+
+# Child class
+class Dog < Animal
+  def speak
+    "Woof!"
+  end
+end
+
+dog = Dog.new
+puts dog.speak # Output: Woof!
+```
+
+## Access Control
+
+Ruby provides three levels of method access control: `public`, `private`, and `protected`.
```

Comment:
Similarly, this may be a better fit for when you hover over `private`, `public` and `protected` as to not make this document too overwhelming.